### ISSUE TITLE
Rewrite notifications to be more flexible/configurable

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#2693]
Opening this because @spencern has mentioned several times that we need to do this and he's very right so I wanted to get this spec'd out so it could be worked on when appropriate.

The current notification system does a good job sending Order notifications, but needs to be refactored to be more general and allow configuring a notification for almost any system event, plus allow which events get SMS and in-app notifications. The current events are essentially hard-coded.

My initial proposal for how this should work would be:

1. Settings panel searches for all available system events (Hooks) and allows you select which events should receive either an in-app notification and/or an SMS (maybe also Slack could go here also?). This should be dynamic so as system events are added they are automatically appear in this settings panel. Probably with a default set for basic Order notification similar to what we have now.
1. You should also be able to configure the content of the message similar to how you can do now with emails (via templates probably?).
1. Should be extendable with other system notification types. (Hopefully Slack could be an included example of this?)
1. Should listen for all system events, check configuration, and then send appropriate notification.
1. Optionally could listen for Errors and allow you to send notification when certain types of errors appear.

Sort of a "sub-task" of #429 ( as was the original notifications system I suppose)

**Comment by zenweasel** [comment]
@spencern Yes, agreed that it makes sense to fold Email into this as just another type of notification

**Comment by spencern** [comment]
We've had suggestion that Browser notifications be considered as part of our notification framework as well.

**Comment by spencern** [comment]
Another issue that may not be directly resolvable by this issue is that order notifications are currently hardcoded to use the ShopId associated with the order. Order notifications (and probably other types of notifications as well) need to be flexible enough to notify users of Shops with order items rather than just the shopId associated with the order.
